From 53a00b25087dbe72f70cb0fa981b4cd9d7c9d15a Mon Sep 17 00:00:00 2001
From: vjudin <vjudin@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 12 Nov 2013 09:33:58 +0000
Subject: [PATCH 0495/1077] Mantis Id:0029699 SNMP service bound by default at
 address 0.0.0.0 (INADDR_ANY) because DHCP server has not still assigned the
 proper IP address at the moment the service starts.

git-svn-id: svn://localhost/SelfServiceCommon/trunk@613 90b65887-3827-0410-9a23-83215b262276
---
 .../cpp/Installer_CustomAction/src/MConfig_CustomAction.cpp  | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/SelfServiceCommon/Massai/cpp/Installer_CustomAction/src/MConfig_CustomAction.cpp b/SelfServiceCommon/Massai/cpp/Installer_CustomAction/src/MConfig_CustomAction.cpp
index 6b2de562..f5c98b05 100644
--- a/SelfServiceCommon/Massai/cpp/Installer_CustomAction/src/MConfig_CustomAction.cpp
+++ b/SelfServiceCommon/Massai/cpp/Installer_CustomAction/src/MConfig_CustomAction.cpp
@@ -156,6 +156,10 @@ extern "C" UINT __stdcall PrepareAddressValues(MSIHANDLE hInstall)
 #ifndef MAXHOSTNAMELEN 
 #define MAXHOSTNAMELEN 512
 #endif
+#if 1
+	static const char* const szInaddrAny = "udp:0.0.0.0:161";
+	hResult = WcaSetProperty( L"SNMPAGENTS", ToWiden( szInaddrAny ).c_str() );
+#else	
     static const size_t namelen (MAXHOSTNAMELEN );
     char szHostName[ MAXHOSTNAMELEN ];
     int rc = ::gethostname( szHostName, MAXHOSTNAMELEN );
@@ -212,6 +216,7 @@ extern "C" UINT __stdcall PrepareAddressValues(MSIHANDLE hInstall)
         wostr << wsFunction << "\tgethostname() failed: rc = " << rc << " WSAGetLastError() = " << WSAGetLastError();
         Log_Str( wostr );
     }
+#endif	
     UINT er = SUCCEEDED(hResult) ? ERROR_SUCCESS : ERROR_INSTALL_FAILURE;
 	wostr << wsFunction << L" er = " << ((er == ERROR_SUCCESS ) ? L"ERROR_SUCCESS" : L"ERROR_INSTALL_FAILURE" );
 	Log_Str( wostr );
-- 
2.41.0.windows.1

